portfolio/src/main/webapp/genre_constants.js

```diff
@@ -0,0 +1,50 @@
+const hipHop = ["Check yo self - ice cube", "Tierra Whack - only child", "Gold teeth - blood orange", "Who dat boy - tyler, the creator", "Poetic justice - kendrick lamar", "Solo (reprise) - frank ocean", "Bank - earthgang", "It ain’t hard to tell - nas", "Jazz - mick jenkins", "Boogie - brockhampton", "Ready or not - fugees", "Wax on - injury reserve", "King kunta - kendrick lamar", 
```

Comment:
I should have raised this up in an earlier review, but better late than never 😬:

Looks like we could use a better data structure here rather than two lists for each genre. In general, we want to make sure that the design of the program is reflected in the code as directly as possible (this is known as then embedded design principle).

Also, Two lists allows for invalid designs to be expressed: e.g. what if the ID list for hiphop is longer than the title list? (This is known as the Representable/valid principle)

Reply:
Got it, that makes a lot of sense, thank you! do you think a JS object would make more sense for this?

Reply:
This can also be represented as hierarchal data e.g (json format)
[
{
   "genreType":"hiphop",
   "values":[
      {
         "id":"HdOhwPovkHI",
         "genre":"Check yo self - ice cube"
      },
      {
         "id":"CXdTShfFPYM",
         "genre":"Tierra Whack - only child"
      }
   ]
},
{
   "genreType":"rAndB",
   "values":[
      {
         "id":"lnUV4cz0gv8",
         "genre":"Cranes in the Sky - Solange"
      },
      {
         "id":"OxdCR6wwoXs",
         "genre":"Diddy Bop - Noname"
      }
   ]
}
]

Then you can take the input file and parse the json data using
result = JSON.Parse(input)
result object will be list of genere types. Each GenereType will have multiple geners and their corresponding id.'

In your developer console:

var a = '[ { "genreType":"hiphop", "values":[ { "id":"HdOhwPovkHI", "genre":"Check yo self - ice cube" }, { "id":"CXdTShfFPYM", "genre":"Tierra Whack - only child" } ] }, { "genreType":"rAndB", "values":[ { "id":"lnUV4cz0gv8", "genre":"Cranes in the Sky - Solange" }, { "id":"OxdCR6wwoXs", "genre":"Diddy Bop - Noname" } ] } ]'

result = JSON.parse(a)

This is the basics of representing hierarchal data and how to parse and consume it. This does not get you type safety out of the box, i.e the hierarchal data might not be in the format you expect. There are ways to add type safety to this using schema etc. But I will ignore this in first pass (if I were you, can be extended in later iterations)

https://www.w3schools.com/js/js_json_intro.asp

Reply:
I made use of object classes by iterating through the arrays, but I ended up keeping the arrays so as to not have to manually create the song objects -- is there a way that I can quickly get rid of the id and song title arrays all together, or would I just have to go through one by one?

Reply:
one way could be to do the conversion in a developer console by importing all the arrays, doing the zip, and printing the resulting json which can be used directly in your code.

Reply:
Done in the latest commit!

Reply:
nice, thank you!